learning_wavelets/data/datasets.py

```diff
@@ -11,26 +11,33 @@ def normalise(image):
     return (image / 255) - 0.5
 
 # data augmentation
-def random_rotate_image(image):
+def random_rotate_flip_image(image):
     # in numpy
-    image = ndimage.rotate(image, np.random.uniform(-30, 30), reshape=False)
+
+    rot_coef = np.random.randint(4, size=1)[0]  # 0-3
+    flip_coef = np.random.randint(2, size=1)[0]  # 0-1
+
+    image = np.rot90(image, rot_coef).copy()
```

Comment:
why do you need the copy ?

Reply:
I got inspired from [here](https://github.com/SonghyunYu/DIDN/blob/master/gray_dataset.py).

Reply:
sure but you need to be critical of what you copy-paste, i.e. understand why this was here in the first place. Here it's probably not going to be a big problem because we prefetch, but generally copy operations are costly, so you want to make sure you understand why you do them.

Reply:
changed